hub/subscribe.go

```diff
@@ -17,41 +17,60 @@ func (h *Hub) SubscribeHandler(w http.ResponseWriter, r *http.Request) {
 		panic("The Response Writer must be an instance of Flusher.")
 	}
 
-	subscriber, updateChan, ok := h.initSubscription(w, r)
+	subscriber, pipe, ok := h.initSubscription(w, r)
 	if !ok {
 		return
 	}
 	defer h.cleanup(subscriber)
 
-	for {
-		if h.options.HeartbeatInterval == time.Duration(0) {
+	if h.options.HeartbeatInterval == time.Duration(0) {
+		for {
 			// No heartbeat defined, just block
-			serializedUpdate, open := <-updateChan
-			if !open {
-				return
+			update, err := pipe.Read()
+			if err != nil {
+				if err == ErrClosedPipe {
+					return
```

Comment:
`close(updates)`? Maybe that this code could be factorized with the similar one below?

Reply:
the chan `updates` does not exists here. There is no race with heartbeat, we can directly read from block `Read` method.